util.py

```diff
@@ -0,0 +1,151 @@
+        self.cursize = 0
+        self.currealsize = 0
+        self.maxsize = maxsize
+        self.tooMuch = False
+        self.done = False
+
+    def write(self, s):
+        size = len(s)
+        self.currealsize += size
+        if self.currealsize > self.maxsize:
+            self.tooMuch = True
+            return
+        else:
+            self.cursize += size
+            return io.StringIO.write(self, s)
+
+    def set_done(self):
+        """Indicate that the reader should stop processing the input given via
+        procces
+        """
+
+        self.done = True
+
+    def getvalue(self):
+        """Return the value that has been written the text
+        "ERROR:    TRUNCATED X bytes ignored" is added
+
+        Returns:
+            string -- The written text as utf-8 string
+        """
+
+        if self.tooMuch:
+            return (io.StringIO.getvalue(self) +
+                    "\n\n" +
+                    "ERROR:    TRUNCATED {} bytes ignored".format(self.currealsize - self.cursize))
+
+        return io.StringIO.getvalue(self)
+
+    def procces(self, stream):
+        """Continiously read stream in thread until no more data comes trough
+        or the procces in closed by calling set_done(). A deamon is used to
+        ensure that swipl does not hang because the output is not read.
+
+        We read per 100 bytes. We try to prevent unicode errors from cropping
+        up during the execution. If our best effort fails, we ignore unknown
+        unicode chars.
+
+        Arguments:
+            stream {file} -- something that is `.read` able
+
+        Returns:
+            SilentLimitedBuffer -- self, for chaining
+        """
+
+        def write_deamon(buff, stream):
+            danglingBytes = None  # will almost alway be None
+
+            while not buff.done:
+                bytedata = stream.read(100)
+                if not bytedata:
+                    break  # no more data
+
+                # If some byte were not parsed in the previous run, prepend
+                # them
+                if danglingBytes:
+                    bytedata = danglingBytes + bytedata
+                    danglingBytes = None
+
+                data = None
+                # Errors might be thrown when a multibyte char is chopped
+                try:
+                    data = bytedata.decode("utf-8")
+                    danglingBytes = None
+                except UnicodeDecodeError:
+                    for i in [-1, -2, -3]:
+                        try:
+                            data = bytedata[:i].decode("utf-8")
+                            danglingBytes = bytedata[:i]
+                            break
+                        except UnicodeDecodeError:
+                            pass
+                    else:
+                        # all failed (no break): ignore errors
+                        data = bytedata.decode("utf-8", errors="ignore")
+                        danglingBytes = None
```

Comment:
This is has been added (and some comments).

An aternative solution would be to just always use `bytedata.decode("utf-8", errors="ignore")` and this might not be so bad as this is only used to render errors in the output.

```suggestion
            while not buff.done:
                bytedata = stream.read(100)
                if not bytedata:
                    break  # no more data

                data = bytedata.decode("utf-8", errors="ignore")
```